### PR TITLE
Fixing SetUp for SLT tests of ShapeOF

### PR DESCRIPTION
**Pull request by nsemaev**
### Details:
 - *Removed passing output_type to the ShapeOf class constructor, it is not needed here, the default value for the argument is described in the class*
 - *ShapeOfLayerTest was removed from the skip config for CPU SLT*
 - *Added a variant of the test with FP32 precision on the input*

### Tickets:
 - *74414*

### CODE REVIEW

File: src/tests/functional/shared_test_classes/src/single_layer/shape_of.cpp [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace LayerTestsDefinitions {
         auto inType = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(inputPrecision);
         auto param = ngraph::builder::makeParams(inType, {inputShapes});
         auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::opset3::Parameter>(param));
-        auto shapeOf = std::make_shared<ngraph::opset3::ShapeOf>(paramOuts[0], inType);
+        auto shapeOf = std::make_shared<ngraph::opset3::ShapeOf>(paramOuts[0]);
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/single_layer/shape_of.cpp:28]: you should add additional parameter, not to remove `inType`. Please, check the bug description again: `std::tie(inputPrecision, outType, inputShapes, targetDevice) = this->GetParam();`
...
`auto shapeOf = std::make_shared<ngraph::opset3::ShapeOf>(paramOuts[0]);`

**Review by iefode — CHANGES_REQUESTED** [review]

File: src/tests/functional/shared_test_classes/src/single_layer/shape_of.cpp (hunk 1 above)
Comment (nsemaev) [reply src/tests/functional/shared_test_classes/src/single_layer/shape_of.cpp:28]: The constructor of the ngraph::opset3::ShapeOf class has the output_type argument, which is equal to element::i64 by default. 

https://github.com/openvinotoolkit/openvino/blob/2f9c5df2717db5f37768c3230749ace79236dc4b/src/core/include/openvino/op/shape_of.hpp#L19

In my opinion this is correct logic and there is no need to pass a custom value in the SetUp() method in the test framework.

Comment (nsemaev) [reply src/tests/functional/shared_test_classes/src/single_layer/shape_of.cpp:28]: The constructor of the ngraph::opset3::ShapeOf class has the output_type argument, which is equal to element::i64 by default. 

https://github.com/openvinotoolkit/openvino/blob/2f9c5df2717db5f37768c3230749ace79236dc4b/src/core/include/openvino/op/shape_of.hpp#L19

In my opinion this is correct logic and there is no need to pass a custom value in the SetUp() method in the test framework.

**Review by iefode — APPROVED** [review]

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/937

**Pull request merged**